coderd/workspacebuilds.go

```diff
@@ -1187,6 +1197,17 @@ func (api *API) buildTimings(ctx context.Context, build database.WorkspaceBuild)
 		})
 	}
 	for _, t := range agentScriptTimings {
+		// Ref: #15432: agent script timings must not have a zero start or end time.
+		if t.StartedAt.IsZero() || t.EndedAt.IsZero() {
+			api.Logger.Debug(ctx, "ignoring agent script timing with zero start or end time",
+				slog.F("workspace_id", build.WorkspaceID),
+				slog.F("workspace_agent_id", t.WorkspaceAgentID),
+				slog.F("workspace_build_id", build.ID),
+				slog.F("workspace_agent_script_id", t.ScriptID),
+			)
+			continue
+		}
+
```

Comment:
The issue referenced in #15432 originates from the `codersdk.TimingStageConnect` timing that's further, but that one doesn't seem addressed?

It seems trying to use `agent.FirstConnectedAt.Time` at the time the build completes is an oversight as the agent may not have connected yet, it should probably be done elsewhere.

Reply:
It turns out the timings chart is built to render with a "loading" state until all of provisioner timings, script timings, and connection timings are populated with at least one value. The assumption here is that timings are only useful after a build has completed.

My previous change (https://github.com/coder/coder/pull/18058) adds a check in the UI so that causes a time range with a zero time to be rendered as an "instant" with an "Invalid" label.

I think that, for now, it makes sense to simply exclude the agent connection timing if `!FirstConnectedAt.Valid`. Does that sound alright?

Reply:
I'm fine with that if it can show up later if the agent finally connects 👍🏻